static/js/src/search-filters.js

```diff
@@ -0,0 +1,64 @@
+/**
+  Update the filter query strings.
+  @param {String} type the type of filter to change.
+  @param {String} value the new filter value.
+*/
+function _searchChangeFilter(type, value) {
+  const url = window.location.href.split('?')[0];
+  const queries = {};
+  window.location.search.substr(1).split('&').forEach(ele => {
+    const parts = ele.split('=');
+    if (parts[0]) {
+      queries[parts[0]] = parts[1];
+    }
+  });
+  // No need to change the url if the filter value is already selected.
+  if (value !== queries[type]) {
+    queries[type] = value;
+    const queryString = Object.keys(queries).map(key => `${key}=${queries[key]}`).join('&');
+    window.location = `${url}?${queryString}`;
+  }
+}
+document.querySelector('.js-sort-select')
+  .addEventListener('change', e => _searchChangeFilter('sort', e.target.value));
+
+function toggleMenu(element, show) {
+  element.setAttribute('aria-expanded', show);
+  element.classList.toggle('p-contextual-menu__toggle--active', show);
+  document
+    .querySelector(element.getAttribute('aria-controls'))
+    .setAttribute('aria-hidden', !show);
+}
+
+function setupContextualMenuListeners(contextualMenuToggleSelector) {
+  const toggles = document.querySelectorAll(contextualMenuToggleSelector);
+  toggles.forEach(toggle => {
+    toggle.addEventListener('mousedown', e => {
+      e.preventDefault();
+      const target = e.target;
+      target.blur();
+      document.querySelector('.p-contextual-menu__dropdown').focus();
+      const menuAlreadyOpen = target.getAttribute('aria-expanded') === 'true';
+      toggleMenu(target, !menuAlreadyOpen);
+    });
+  });
+  document.addEventListener('click', e => {
+    toggles.forEach(toggle => {
+      const contextualMenu = document.querySelector(toggle.getAttribute('aria-controls'));
+      const clickOutside = !(toggle.contains(e.target) || contextualMenu.contains(e.target));
+      if (clickOutside) {
+        toggleMenu(toggle, false);
+      }
+    });
+  });
+  document.onkeydown = e => {
```

Comment:
It's possible that some other script on the page has already set an `onkeydown` function and this will be overriding that. You might want to do something like....

```suggestion
  var oldKeyDown = document.onkeydown;
  document.onkeydown = e => {
    e = e || window.event;
    if (e.keyCode === 27) {
      toggles.forEach(toggle => {
        toggleMenu(toggle, false);
      });
    }
    if (oldKeyDown) {
      oldKeyDown(e);
    }
  };
```

Reply:
Done.